presentation/testing/src/presentation-testing/IModelUtilities.ts

```diff
@@ -0,0 +1,96 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module IModel
+ */
+
+import { IModelDb, IModelJsFs, SnapshotDb } from "@itwin/core-backend";
+import { Id64String } from "@itwin/core-bentley";
+import { BisCodeSpec, Code, CodeScopeProps, CodeSpec, ElementAspectProps, ElementProps, LocalFileName, ModelProps } from "@itwin/core-common";
+import { IModelConnection, SnapshotConnection } from "@itwin/core-frontend";
+import path from "path";
+import { getTestOutputDir } from "./Helpers";
+
+/**
+ * Interface for IModel builder pattern. Used for building IModels to test rulesets.
+ *
+ * @internal
+ * @beta
+ */
+export interface TestIModelBuilder {
+  /** Insert a model into the builder's iModel */
+  insertModel<TProps extends ModelProps>(props: TProps): Id64String;
+  /** Insert an element into the builder's iModel */
+  insertElement<TProps extends ElementProps>(props: TProps): Id64String;
+  /** Insert an element aspect into the specified element */
+  insertAspect<TProps extends ElementAspectProps>(props: TProps): void;
+  /** Create code for specified element */
+  createCode(scopeModelId: CodeScopeProps, codeSpecName: BisCodeSpec, codeValue: string): Code;
+}
+
+/**
+ * Function that takes builder actions and returns a connection to the build database.
+ * @param name Name of test IModel
+ * @param cb Callback function that executes all given builder actions
+ *
+ * @internal
```

Comment:
Aren't we planning for this to be public (beta for the time being)?

Reply:
Changed to `@beta`